File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var removeOuterParentheses = function(s) {
    let opened = 0
    let res = ""
    for (const c of s) {
        if (c === "(" && opened++ > 0) res += c
        if (c === ")" && opened-- > 1) res += c
    }
    return res
};

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        res, opened = "", 0
        for c in s:
            if c == "(" and opened > 0:
                res += c
            if c == ")" and opened > 1:
                res += c
            opened += 1 if c == "(" else -1
        return res